src/webots/core/WbDesktopServices.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright 1996-2019 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "WbDesktopServices.hpp"
+#include <QtCore/QProcess>
+
+#ifndef __linux__
+#include <QtGui/QDesktopServices>
+#endif
+
+bool WbDesktopServices::openUrl(const QString &url) {
+#ifdef __linux__
+  QProcess process;
+  process.setProgram("xdg-open");  // we need to use xdg-open to be snap compliant
+  process.setArguments(QStringList() << url);
+  return process.startDetached();
+#else
+  return QDesktopServices::openUrl(url);
```

Comment:
Compilation error on Windows: `QDesktopServices::openUrl` expects a `const QUrl &` argument and not a `const QString &` argument.

probably the best solution would be to change it to:
```suggestion
  return QDesktopServices::openUrl(QUrl(url));
```
and to fix the calls of WbDesktopServices::openUrl() to remove when possible the duplicated `openUrl(QUrl(<string>).toString())` instructions.
For example in https://github.com/cyberbotics/webots/blob/e837ffa64a7361706a2d183e0763466ae931ddca/src/webots/core/WbFileUtil.cpp#L273

Reply:
`WbDesktopServices::openUrl()` should receive a `QString` argument to work with the `connect(label, &QLabel::linkActivated, &WbDesktopServices::openUrl);` connections. So we should probably keep it like that (or create new method only for the connect callbacks that would convert `QString` to `QUrl` and call WbDesktopServices::openUrl()`.

Reply:
Yes, I don't suggested to change the `WbDesktopServices::openUrl()` function, but to change when possibible the use of this function from `openUrl(QUrl(<string>).toString())` to ``openUrl(<string>))

Reply:
Ok, I see what do you mean.
So then I would at least remove a duplicated interpretation of the url:

```suggestion
  return QDesktopServices::openUrl(QUrl(url, QUrl::StrictMode));
```

Reply:
For some reason, I cannot commit your suggestion from this page (the popup menu is disabled). I will commit from git.

Reply:
It seems it also needed to include `QtCore/QUrl`.

Reply:
All this should be fixed with my last commit.